src/pages/MyMove/PPMBooking/DateAndLocation.test.jsx

```
@@ -0,0 +1,101 @@
+    moveTaskOrderID: mockMoveId,
+    ppmShipment: {
+      pickupPostalCode: '20002',
+      secondaryPickupPostalCode: '20003',
+      destinationPostalCode: '20004',
+      secondaryDestinationPostalCode: '20005',
+      sitExpected: true,
+      expectedDepartureDate: '2022-12-31',
+    },
+  },
+};
+
+describe('DateAndLocation component', () => {
+  describe('creating a new PPM shipment', () => {
+    it('renders the heading and empty form', () => {
+      render(<DateAndLocation {...defaultProps} />);
+
+      expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('PPM date & location');
+    });
+
+    it('routes back to the new shipment type screen when back is clicked', async () => {
+      const selectShipmentType = generatePath(customerRoutes.SHIPMENT_SELECT_TYPE_PATH, {
+        moveId: mockMoveId,
+      });
+
+      render(<DateAndLocation {...defaultProps} />);
+
+      const backButton = await screen.getByRole('button', { name: 'Back' });
+      userEvent.click(backButton);
+
+      expect(mockPush).toHaveBeenCalledWith(selectShipmentType);
+    });
+  });
+
+  describe('editing an existing PPM shipment', () => {
+    it('renders the heading and form with shipment values', async () => {
+      render(<DateAndLocation {...defaultProps} {...fullShipmentProps} />);
+
+      expect(screen.getByRole('heading', { level: 1 })).toHaveTextContent('PPM date & location');
+
+      const postalCodes = screen.getAllByLabelText('ZIP');
+      const secondaryPostalCodes = screen.getAllByLabelText('Second ZIP');
+
+      await waitFor(() => {
+        expect(postalCodes[0]).toHaveValue('20002');
+        expect(postalCodes[1]).toHaveValue('20004');
+        expect(secondaryPostalCodes[0]).toHaveValue('20003');
+        expect(secondaryPostalCodes[1]).toHaveValue('20005');
+        expect(screen.getAllByLabelText('Yes')[2]).toBeChecked();
+        expect(screen.getByLabelText('When do you plan to start moving your PPM?')).toHaveValue('31 Dec 2022');
```

I think it's suggested that we only have the main necessary `expect` statement in the `await waitFor` block, since one is available, the rest should be too, and if not, we can fail faster: https://kentcdodds.com/blog/common-mistakes-with-react-testing-library#having-multiple-assertions-in-a-single-waitfor-callback